RadixWallet/Features/CardCarousel/CardCarousel+View.swift

```diff
@@ -0,0 +1,194 @@
+import SwiftUI
+
+// MARK: - CardCarousel.View
+extension CardCarousel {
+	public struct View: SwiftUI.View {
+		public let store: StoreOf<CardCarousel>
+
+		private let margin: CGFloat = .medium1
+		private let spacing: CGFloat = .small1 * 0.5
+		@ScaledMetric private var height: CGFloat = 105
+		@SwiftUI.State private var selectedCardIndex = 0
+
+		public var body: some SwiftUI.View {
+			WithPerceptionTracking {
+				VStack(spacing: 8) {
+					GeometryReader { _ in
+						coreView
+					}
+					.frame(height: store.cards.isEmpty ? 0 : height)
+
+					positionIndicator
+				}
+			}
+			.onAppear {
+				store.send(.view(.didAppear))
+			}
+		}
+
+		@MainActor
+		private var coreView: some SwiftUI.View {
+			TabView(selection: $selectedCardIndex) {
+				ForEach(Array(store.cards.enumerated()), id: \.1) { index, card in
+					CarouselCardView(card: card) {
+						store.send(.view(.cardTapped(card)))
+					} closeAction: {
+						store.send(.view(.closeTapped(card)), animation: .default)
+					}
+					.tag(index)
+					.padding(.horizontal, margin - spacing)
+					.transition(.scale(scale: 0.8).combined(with: .opacity))
+				}
+			}
+			.tabViewStyle(.page(indexDisplayMode: .never))
+			.animation(.default, value: store.cards)
+		}
+
+		private var positionIndicator: some SwiftUI.View {
+			HStack(spacing: spacing) {
+				if store.cards.count > 1 {
+					ForEach(0 ..< store.cards.count, id: \.self) { index in
+						let isSelected = selectedCardIndex == index
+						Capsule()
+							.fill(isSelected ? .app.gray4 : .app.gray2)
+							.frame(isSelected ? spacing : .small3)
+					}
+				}
+			}
+		}
+	}
+}
+
+// MARK: - CarouselCardView
+public struct CarouselCardView: View {
+	public let card: HomeCard
+	public let action: () -> Void
+	public let closeAction: () -> Void
+
+	public var body: some View {
+		ZStack(alignment: .topTrailing) {
+			Button(action: action) {
+				VStack(alignment: .leading, spacing: .small2) {
+					HStack {
+						Text(title)
+							.textStyle(.body1Header)
+							.minimumScaleFactor(0.8)
+
+						if showLinkIcon {
+							Image(.iconLinkOut)
+								.foregroundStyle(.app.gray2)
+						}
+					}
+
+					Text(text)
+						.lineSpacing(-20)
+						.lineLimit(nil)
+						.minimumScaleFactor(0.8)
+						.textStyle(.body2Regular)
+				}
+				.multilineTextAlignment(.leading)
+				.foregroundStyle(.app.gray1)
+				.padding([.top, .leading], .medium2)
+				.padding(.trailing, trailingPadding)
+				.padding(.bottom, .small1)
+				.frame(maxWidth: .infinity, maxHeight: .infinity, alignment: .topLeading)
+				.background(alignment: .trailing) {
+					background
+				}
+				.background(.app.gray5)
+				.cornerRadius(.small1)
+			}
+
+			Button(action: closeAction) {
+				Image(asset: AssetResource.close)
+					.resizable()
+					.frame(width: .medium3, height: .medium3)
+					.tint(.app.gray1)
+					.padding(.small2)
+			}
+			.frame(.small, alignment: .topTrailing)
+		}
+	}
+
+	private var trailingPadding: CGFloat {
```

Comment:
Not quite getting what this is for, I guess I need to play with it.

Reply:
basically how much space we must leave empty from the trailing side of the view. Depending on its content, we have more or less space that we can use without the text overlapping with an image.

Original implementation had a fixed value, but that would leave very little space for some views on smaller devices (as iPhone SE) and content would look trimmed.

Reply:
The reason it was fixed is that the image we want to avoid covering is fixed size, because the view has a fixed height. That in turn is because tabview can't handle flexible heights very well.